crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,456 @@
+//! Components used to control the layout of UI-[`Node`](super::Node) entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+/// Grouping of core control parameter of the layout for this node.
+#[derive(Component, Copy, Clone, PartialEq, Debug, Default, Serialize, Deserialize, Reflect)]
+#[reflect(PartialEq, Serialize, Deserialize)]
+pub struct LayoutControl {
+    /// Defines how the node will parttake in the layouting.
+    pub display: Display,
+    /// The strategy used to position this node.
+    pub position: Position,
+    /// The inset of this UI node, relative to its default position
+    pub inset: Inset,
+    /// The behavior in case the node overflows its allocated space
+    pub overflow: Overflow,
+}
+
+/// Defines how the node will parttake in the layouting.
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect)]
+#[reflect(PartialEq, Serialize, Deserialize)]
+pub enum Display {
+    /// Use Flexbox layout model to determine the position of this [`Node`](crate::ui_node::Node).
+    #[default]
+    Flex,
+    /// The node is completely removed from layouting.
+    ///
+    /// The final layout will be calculated as if this node never existed.
+    /// If you want to hide a node and its children, but keep its layout in place,
+    /// set its [`Visibility`](bevy_render::view::Visibility) component instead.
+    None,
+}
+
+/// The strategy used to position this node.
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Position {
+    /// Positioned in relation to its parent, while considering all sibling-nodes that also have the [`Position::Relative`] value.
+    #[default]
+    Relative,
+    /// Positioned in relation to its parent, without considering any sibling nodes.
+    Absolute,
+}
+
+/// The size of the gutters between the rows and columns of the flexbox layout
+///
+/// Values of `Size::UNDEFINED` and `Size::AUTO` are treated as zero.
+#[derive(Copy, Clone, PartialEq, Debug, Default, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct Gap(pub Size);
```

Comment:
Now that gap is nested under `FlexContainer` I believe this `Gap` struct should be removed, and the `gap` property should just be a plain `Size`.

Reply:
Good point!